qiita_db/reference.py

```diff
@@ -171,6 +171,10 @@ def tree_fp(self):
                         JOIN qiita.{0} r ON r.tree_filepath=f.filepath_id
                      WHERE r.reference_id=%s""".format(self._table)
             qdb.sql_connection.TRN.add(sql, [self._id])
-            rel_path = qdb.sql_connection.TRN.execute_fetchlast()
+            try:
```

Comment:
I'm confused here. I don't see any new test in the test_reference.py file, and if all the previous unit tests where passing, how is possible that there is a test that it's executing this code?
Also, I don't think the functionality is correct here. If the reference doesn't have a tree, it should return `None`. Currently, it will return the reference mountpoint:
```python
In [6]: import qiita_db as qdb

In [7]: _, basefp = qdb.util.get_mountpoint('reference')[0]

In [8]: rel_path = ''

In [9]: from os.path import join

In [10]: join(basefp, rel_path)
Out[10]: '/Users/jose/qiime_software/QiiTa/qiita_db/support_files/test_data/reference/'
```
Which I think reinforces the idea of having a specific test in which the tree is `Null`.
Furthermore, the approach that we used in cases where the value may not exist is:
```python
rel_path = qdb.sql_connection.TRN.execute_fetchindex()
if rel_path:
    rel_path = rel_path[0][0]
else:
    return None
```